File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java
class Solution {
    public long solution(long n) {
        char[] chars = Long.toString(n).toCharArray();
        int[] arr = new int[10];
        for (char i : chars){
            arr[i - '0']++;
        }
        StringBuilder sb = new StringBuilder();
        for (int i = 9; i >=0;i--){
            for (int j = 0; j < arr[i];j++){
                sb.append(i);
            }
        }
        return Long.parseLong(sb.toString());
    }
}

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.kt
class Solution {
    fun solution(n: Long): Long {
        return buildString {
            val arr = IntArray(10)
            n.toString().forEach {
                arr[it.digitToInt()]++
            }
            for (i in 9 downTo 0){
                repeat(arr[i]){ append(i) }
            }
        }.toLong()
    }
}